akka-http-core/src/main/scala/akka/http/impl/model/parser/UriParser.scala

```diff
@@ -194,7 +220,7 @@ private[http] final class UriParser(
   def pchar = rule { `path-segment-char` ~ appendSB() | `pct-encoded` }
 
   def rawQueryString = rule {
-    clearSB() ~ oneOrMore(`raw-query-char` ~ appendSB()) ~ run(setRawQueryString(sb.toString)) | run(setRawQueryString(""))
+    clearSB() ~ oneOrMore(`query-char` ~ appendSB() | `pct-encoded`) ~ run(setRawQueryString(sb.toString)) | run(setRawQueryString(""))
```

Comment:
So, when parsing a full URI, we first parse the raw query string into the var and then later `Uri.create` is called which will have another parse run using `parseRawQueryString` to fix potentially broken characters. That seems a bit wasteful. Is there a way we could do it in one step?

I guess it's not performance critical if you use `strict` mode but I guess we might need to be careful because URI parsing is one of the hotspots of the whole request processing.